### PR TITLE
PLANET-6638: Issues block pattern

### PR DESCRIPTION
**Pull request by GP-Dan-Tovbein**
Ref: https://jira.greenpeace.org/browse/PLANET-6638

Design: [Figma](https://www.figma.com/file/99X4IXpkQIgNKtKvHEyI1P/IA-%26-nav_MVP-prototype?node-id=31%3A219)

[Demo page](https://www-dev.greenpeace.org/test-sinope/issues-pattern)

**Testing**
Go to the instance > Create new page > Click on "+" > Search Issues pattern

**There are some UI pattern's limitations, ~that is why we add some styles through an external CSS file (see the class deep-dive-topic class)~**

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [sinope](https://www-dev.greenpeace.org/test-sinope) | [admin](https://www-dev.greenpeace.org/test-sinope/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-sinope/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-sinope) | [composer-local.json](https://github.com/greenpeace/planet4-test-sinope/blob/main/composer-local.json)

:watch: 2022.03.23 11:32:45

File: classes/patterns/class-issues.php
Comment (mleray) [inline classes/patterns/class-issues.php:55]: We should add translation strings for placeholders, you can see [how I added them for the Highlighted CTA pattern](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/814/files#diff-5c0d4b33eaa8eb18a44ba1f1e0fb04b29c8c3492b6b4dfdc95f5dd4edeaf8c26R44)

**Comment by mleray** [comment]
Do we need to initialise it with so many columns? 🤔 On my local it looks a bit weird in the editor:

Screenshot 2022-03-24 at 11 45 48

File: assets/src/styles/blocks/Issues.scss
Comment (mleray) [inline assets/src/styles/blocks/Issues.scss:1]: Ideally, we should avoid adding custom classes to the patterns, because the second it gets added to a page it's basically just a block... We can style the root block (for example the WP core `Columns` block, I created a [ticket](https://jira.greenpeace.org/browse/PLANET-6683) to get design recommendations for it), but then it will be styled for all occurrences of said block, not just the one used in the pattern. It's actually better for consistency too.
There are some native ways to style the blocks that are used in the pattern:
- You can usually define color and background color in the attributes when initialising a block (such as `Media & Text`).
- For spacing, it's nice to use the `Spacer` block or [Bootstrap classes](https://getbootstrap.com/docs/5.0/utilities/spacing/) that we already use in other places.

It is less customisable than adding a custom class, so it means sometimes we'll have to compromise, but I think it makes more sense for patterns 🙂

File: assets/src/styles/blocks/DeepDiveTopic.scss
Comment (GP-Dan-Tovbein) [inline assets/src/styles/blocks/DeepDiveTopic.scss:1]: I need to add this styles since the editor does not provide some of them, like `box-shadow`, `min-height`, among others.

c.c. @mleray

Comment (comzeradd) [reply assets/src/styles/blocks/DeepDiveTopic.scss:1]: We can leave this customizations out of this implementation for now, if there is no other way of applying them, and let the design team know. Most of these patterns are under assessment as pilots will start using them. So better not add any specific css classes just yet.

File: classes/patterns/class-issues.php
Comment (mleray) [inline classes/patterns/class-issues.php:32]: Those keywords need to be translated 🤔 Also, who decided on which ones to add?

File: classes/patterns/class-issues.php
Comment (mleray) [reply classes/patterns/class-issues.php:65]: Also, maybe you don't know this (I sure didn't 😬) but you can actually use `rem` values instead of `px` for the `fontSize` attribute, I'm not sure about this specific case but it might be better?

File: classes/patterns/class-issues.php
Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:32]: > who decided on which ones to add?
Nobody, I've added them since these are the used blocks 😶

Comment (mleray) [reply classes/patterns/class-issues.php:32]: Or we could ask the design team if they have any UX/UI recommendations for this

**Review by mleray — CHANGES_REQUESTED** [review]
It seems that there is an error somewhere in the markup, when testing on local I get this when inserting a new Issues pattern:
Screenshot 2022-04-13 at 14 22 23

File: classes/patterns/class-issues.php
Comment (mleray) [inline classes/patterns/class-issues.php:63]: We should now use the core class `alignfull` class instead of `block-wide` (see this [previously merged PR](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/820)). I think that you should also put the `block` class since [it adds our spacing values](https://github.com/greenpeace/planet4-master-theme/blob/master/assets/src/scss/layout/_blocks.scss) 🤔

File: classes/patterns/class-issues.php
Comment (Inwerpsel) [inline classes/patterns/class-issues.php:60]: Can we do this without needing 2 groups? The block's complexity would be already a bit more manageable with 1 level less.

Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: The first group is the wide block that needs to be fit to the screen's width. And the second group is `container` class.

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: Yeah this one makes sense and doesn't complicated it that much.

It's mostly a problem in the row of items below, as there the complexity is repeated for every element. That makes operations like selecting the right block in the editor very annoying. The paragraphs are currently 8 levels deep. While for a user it seems like the block would only need 3 or 4.
Screenshot from 2022-04-19 22-02-00

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: I created the [same block using the "Row" block](https://www-dev.greenpeace.org/gutenberg/wp-admin/post.php?post=27114&action=edit) (which is enabled because on the gutenberg instance we don't severely limit the core block functionality).
With just 1 additional CSS rule, it looks identical to what's currently in the PR.
Screenshot from 2022-04-19 22-02-29

Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: I completely agree to use `Row` instead of `Column`, in fact that was my initial plan. But the problem was that we needed -at least- a bit customization and it wasn't what we wanted to achieve. 

If we're fine to go with a bit customization then I suggest to use this [deleted file](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/811/commits/a917eaf2373318fa8cacc1cc27aac36a7c4696b6#diff-ac498fa0f16734d2b3a1af452585c6f5828b5612676c2e2b6c6dddc0409781b5) (with a couple of changes) and then provide to the NRO, the CSS in case they need it just by adding the classname through the editor panel (additional settings). 

What do you think?

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: > But the problem was that we needed -at least- a bit customization and it wasn't what we wanted to achieve.

Can you elaborate? Which customization, and why "wasn't it what we wanted to achieve"?

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: > If we're fine to go with a bit customization then I suggest to use this deleted file (with a couple of changes) and then provide to the NRO, the CSS in case they need it just by adding the classname through the editor panel

In my example you see I only had to add 1 CSS rule to set a out padding on the media and text block. That seems a pretty common use case, so we could register it as a block style. Then we could use it every time instead of nesting blocks to achieve padding.

**Comment by Inwerpsel** [comment]
@GP-Dan-Tovbein Isn't each item supposed to link to a page?

File: classes/patterns/class-issues.php
Comment (Inwerpsel) [inline classes/patterns/class-issues.php:61]: A separate issue as the previous line, but putting it here to start a new thread.

We should use
```
"align":"full",
```
in the attributes instead of only putting the `.alignfull` class on it. It uses the editor's alignment control for that.
Screenshot from 2022-04-20 13-46-21

File: classes/patterns/class-issues.php
Comment (Inwerpsel) [inline classes/patterns/class-issues.php:60]: I [added a few classes here as a style to media and text](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/832) (and some other core types that don't use any styles yet), and other blocks. https://www-dev.greenpeace.org/test-venus/issues-pattern-with-padding-style/

https://user-images.githubusercontent.com/7604138/164225326-d87c7a86-1456-405f-91ca-cbfbfe9aa7b2.mp4

Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: > Can you elaborate? Which customization, and why "wasn't it what we wanted to achieve"?
For example `box-shadow`, `border-radius`, `font-family`, `margin` between image and text. In fact, I had to add those styles to [this GPBR page](https://www-dev.greenpeace.org/brasil/trabalhe-conosco-2/) because they requested that.

Comment (comzeradd) [reply classes/patterns/class-issues.php:60]: > With just 1 additional CSS rule, it looks identical to what's currently in the PR.

Wouldn't that require a css class explicitly for this pattern?

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: No, we could register the style on the Media and Text block type, it currently doesn't have any block styles. It's often the case when you use this block that it needs padding. For now we had to add additional blocks just to get this padding right. With a style it's just 1 click in the sidebar. See also [this message](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/811#discussion_r854049107).

**Comment by GP-Dan-Tovbein** [comment]
> @GP-Dan-Tovbein Isn't each item supposed to link to a page?

Nope, they aren't links.

File: classes/patterns/class-issues.php
Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: I really like to register styles, but I think that the padding classes [you added here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/832) are very specific to this block. What do you think?

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: You mean this style is not reusable in other cases? :thinking:

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: You can register the same style for multiple blocks. I added that to the PR now, should be on the test instance soon.

Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: > You mean this style is not reusable in other cases?

Yes, but if you want to apply more padding to another `media-text` you should override it.

Comment (Inwerpsel) [reply classes/patterns/class-issues.php:60]: Ah you mean the exact padding values. Well we do have a sort of design system, so we can create some values that are likely to work in a lot of cases.

For now I kept it simple in the PR and set a single padding of 8 / 16 / 32 px. I also included `align-self: stretch` as it seemed safe to do. How do you suggest making this more generic?

Comment (GP-Dan-Tovbein) [reply classes/patterns/class-issues.php:60]: Let's continue the padding conversation on this [PR](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/832/files)

File: admin/js/editor.js
Comment (GP-Dan-Tovbein) [inline admin/js/editor.js:58]: This style allows to editor to choose the `Roboto` font family. Also defined through the design system.

**Review by Inwerpsel — APPROVED** [review]

**Comment by comzeradd** [comment]
@dantovbein make sure to squash commits before merge.

**Comment by GP-Dan-Tovbein** [comment]
@Inwerpsel I've updated the style and it's now rendered correctly through small and medium screens.
Screenshot 2022-05-04 at 18 59 12
Screenshot 2022-05-04 at 18 59 33

Screenshot 2022-05-04 at 18 59 03

**Comment by GP-Dan-Tovbein** [comment]
@Inwerpsel FYI, Houssam has approved the UAT! So, let me know if this PR is fine again!

**Review by Inwerpsel — APPROVED** [review]

**Pull request merged**
